src/StatusQ/Controls/Validators/StatusAddressValidator.qml

```diff
@@ -0,0 +1,15 @@
+import StatusQ.Controls 0.1
+import StatusQ.Core.Utils 0.1
+
+StatusValidator {
+    name: "address"
+
+    error: qsTr("Please enter a valid address")
+
+    validate: function (t) {
+        // TODO: Add ENS name resolution and validation a la
+        // `ui/shared/AddressInput.qml`
```

Comment:
You can remove this comment.

ENS name resolution requires `asyncValidators` which would be a different pipeline (triggered **after** synchronous pipeline. But this doesn't exist yet.

Reply:
Created an issue here: https://github.com/status-im/StatusQ/issues/395

I could imagine we'll end up with a separate validator (`StatusENSValidator`?) which will be a `StatusAsyncValidator` and has to be applied in addition to `StatusAddressValidator` to get the same effect as what we have in legacy `AddressInput` at the moment.

Reply:
Unrelated to asynchronous validators, introducing a separate validator for ENS means the input would have to be a valid address *AND* a valid ENS address (whether or not the resolution to an address happens). But what we really want is validation for `is it an address` *OR* `is it a valid ENS name`. Maybe it would be good to keep both the static (synchronous) checks for address OR ENS in the same validator, but change the name to `StatusAddressOrEnsValidator`.

As mentioned in #395, if `StatusAddressOrEnsValidator` validates, then the asynchronous resolution and validation can happen.

What do you think?

Reply:
I've gone ahead and added a `StatusAddressValidator` and a `StatusAddressOrEnsValidator` which are synchronous, and allow room for async validators discussed in #395.